adapters/repos/db/lsmkv/store.go

```diff
@@ -41,6 +41,7 @@ type Store struct {
 	// Prevent concurrent manipulations to the bucketsByNameMap, most notably
 	// when initializing buckets in parallel
 	bucketAccessLock sync.RWMutex
+	bucketByNameLock map[string]*sync.Mutex
```

Comment:
Would it be easier to just create a lockable struct as the value type for `bucketsByName`? 

```go
type lockableBucket struct {
    *Bucket
    sync.Mutex
}

type Store struct {
    // ...
    bucketsByName map[string]*lockableBucket
    // ...
}
```

Reply:
> Would it be easier to just create a lockable struct as the value type for `bucketsByName`?
> 
> ```go
> type lockableBucket struct {
>     *Bucket
>     sync.Mutex
> }
> 
> type Store struct {
>     // ...
>     bucketsByName map[string]*lockableBucket
>     // ...
> }
> ```

the bucket itself is not what is being protected but its associated position in the `bucketsByName` map, so to prevent data races while doing `bucketsByName[myBucket] = NewBucket(...) and/or  reading bucketsByName[myBucket]`

Reply:
based on your idea, a lockableMap would be interesting